lib/pbench/server/cache_manager.py

```diff
@@ -280,13 +281,12 @@ def extract(self, path: str) -> str:
             The named file as a string
         """
         try:
-            return (
-                tarfile.open(self.tarball_path, "r:*").extractfile(path).read().decode()
-            )
+            return tarfile.open(tarball_path, "r:*").extractfile(path).read().decode()
         except Exception as exc:
-            raise MetadataError(self.tarball_path, exc)
+            raise MetadataError(tarball_path, exc)
 
-    def get_metadata(self) -> JSONOBJECT:
+    @staticmethod
+    def get_metadata(tarball_path: Path) -> JSONOBJECT:
```

Comment:
The way this is currently structured, this could probably be `_get_metadata` to discourage external use.